pkg/placement/raft/fsm_test.go

```diff
@@ -34,10 +34,11 @@ func TestFSMApply(t *testing.T) {
 		}
 
 		resp := fsm.Apply(raftLog)
```

Comment:
should Apply return bool, error instead of interface{}?

Reply:
this is the by-design of hashicorp raft implementation. resp can be error or actual value. so it always needs to check if resp can be type conversion or not. please look at the other code to see how it handles error or bool.